src/checker.rs

```diff
@@ -47,17 +47,35 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
         return false;
     }
 
-    let mut words = trimmed.split_whitespace();
+    let words = trimmed.split_whitespace();
```

Comment:
I'd suspect that the reason this was mutable was for the `words.any()` call to work further down. However I don't think this is the right fix for the underlying problem, so I prefer getting rid of this `mut` here.

While cloning `words` twice needs some memory, this is nowhere close to being an actual problem with the usual sizes of a Wikipedia article or anything else that could be fed to `sentence-extractor`. Of course in theory you could feed it gigabytes of data through a single file, but I'd argue we'd still be fine here.

Reply:
In the original "words" was only used once, so that was not a problem. But with this PR I use it again. So it became a must to make it immutable and clone it as `words.clone().any()`.

AFAIK, the check function is called per sentence, and a sentence should fit 10 sec as of now, so it will not be huge.

Reply:
> AFAIK, the check function is called per sentence, and a sentence should fit 10 sec as of now, so it will not be huge.

Correct, you're right. Even less likely a problem then.